compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Program.ts

```diff
@@ -271,6 +276,137 @@ function isFilePartOfSources(
   return false;
 }
 
+function getExternalFunctions(
```

Comment:
This function is the primary change in this PR. When collecting external functions imports to insert into the program, we also generate uids for each import.

In cases we cannot generate uids (e.g. program shadows global variables), we bail out without modifying the program (see `error.emit-freeze-conflicting-global.js`)